internal/wasm/sdk/internal/wasm/vm.go

```diff
@@ -579,7 +630,19 @@ func (i *VM) RemoveDataPath(ctx context.Context, path []string) error {
 		return err
 	}
 
-	if err := i.free(ctx, pathAddr); err != nil {
+	if err := i.valueFree(ctx, pathAddr); err != nil {
+		return err
+	}
+
+	// Stash free blocks so eval() calls don't leak them when calling setHeapState()
+	err = i.heapBlocksStash(ctx)
+	if err != nil {
+		return err
+	}
```

Comment:
[code style] it's a minor thing, but we've got a mix of styles here when it comes to the scoping of err. I think we could use `if err := ...; err != nil { return err }` wherever it's _possible_.